File: trunk/Monolingual/ProgressWindowController.m
/*
 *  Copyright (C) 2001, 2002  Joshua Schrier (jschrier@mac.com),
 *  2004-2014 Ingmar Stein
 *  Released under the GNU GPL.  For more information, see the header file.
 */

#import "ProgressWindowController.h"

@implementation ProgressWindowController

- (IBAction)cancelButton:(id)sender
{
	[self.applicationText setStringValue:NSLocalizedString(@"Canceling operation...", "")];
	[self.fileText setStringValue:@""];

	[self.window orderOut:sender];
	[NSApp endSheet:self.window returnCode:1];
}

- (void) start
{
	[self.progressBar setUsesThreadedAnimation:YES];
	[self.progressBar startAnimation:self];
	[self.applicationText setStringValue:NSLocalizedString(@"Removing...", "")];
	[self.fileText setStringValue:@""];
}

- (void) stop
{
	[self.progressBar stopAnimation:self];
}

- (void) setFile:(NSString *)file
{
	[self.fileText setStringValue:file];
}

- (void) setText:(NSString *)text
{
	[self.applicationText setStringValue:text];
}

@end

File: trunk/Monolingual/ProgressWindowController.swift
/*
 *  Copyright (C) 2001, 2002  Joshua Schrier (jschrier@mac.com),
 *  2004-2014 Ingmar Stein
 */

import Cocoa

class ProgressWindowController : NSWindowController {
	@IBOutlet var progressBar: NSProgressIndicator
	@IBOutlet var applicationText: NSTextField
	@IBOutlet var fileText: NSTextField

	@IBAction func cancelButton(sender: AnyObject) {
		self.applicationText.stringValue = NSLocalizedString("Canceling operation...", comment:"")
		self.fileText.stringValue = ""

		self.window.orderOut(sender)
		NSApp.endSheet(self.window, returnCode:1)
	}

	func start() {
		self.progressBar.usesThreadedAnimation = true
		self.progressBar.startAnimation(self)
		self.applicationText.stringValue = NSLocalizedString("Removing...", comment:"")
		self.fileText.stringValue = ""
	}

	func stop() {
		self.progressBar.stopAnimation(self)
	}

	func setFile(file: String) {
		self.fileText.stringValue = file
	}

	func setText(text: String) {
		self.applicationText.stringValue = text
	}

}
